### PR TITLE
CLI docs update and markdown linting.

### LINKED ISSUE #3426: CLI Builds not found in expected location
## Environment details:

* deploying to Ubuntu machine (laptop), but struggling with installation steps
* version of docker: `17.12.1-ce`, ansible `2.3.0`, ubuntu: `17.10`

## Steps to reproduce the issue:

1.   Work through the setup instructions as far as https://github.com/lornajane/incubator-openwhisk/blob/master/ansible/README.md#configuring-the-installation-of-wsk-cli
2.   The documented gradle command doesn't work, but the instructions in the CLI repo itself suggest `./gradlew compile -PnativeCompile` which does build something
3.   However the `ansible-playbook -i environments/<environment> openwhisk.yml -e cli_installation_mode=local -e openwhisk_cli_home=../incubator-openwhisk-cli` still fails

```
TASK [cli : ... or Copy release archive to build directory] ***********************************************************
Friday 09 March 2018  22:09:41 +0000 (0:00:00.045)       0:01:18.569 ********** 
fatal: [172.17.0.1]: FAILED! => {"changed": false, "failed": true, "msg": "Unable to find '/home/lorna/projects/incubator-openwhisk-cli/release/OpenWhisk_CLI-latest-all.tgz' in expected paths."}

Unable to find '/home/lorna/projects/incubator-openwhisk-cli/release
/OpenWhisk_CLI-latest-all.tgz' in expected paths.
```

Also if you try to fix something and then repeat the command it will fail until you manually `docker stop` and then `docker rm` the `invoker0` container (this may not be the right fix for the problem but it's allowing my command to run a bit longer!)

## Provide the expected results and outputs:

I expected the CLI to be installed

## Provide the actual results and outputs:

The CLI isn't installed, and I'm not able to proceed further with the installation (suggestions for workarounds are also welcome here)

## Additional information you deem important:

I have the newest code (as of a couple of days ago) and I think this differs from #3329 but could be related to it.

### PR DESCRIPTION
**Pull request by jonpspri**
Update CLI documentation per #3426 to acurately reflect the process for local CLI build.

## Description
Update to `ansible/README.md` to reflect changes in the CLI build process and to clarify the need
to run `mode=clean` as part of attempting an in-place install.  This change alone does not close #3426 as there are still changes needed to the `incubator-openwhisk-cli` documentation.

For @dubeejw @lornajane 

## Related issue and scope

- [X] I opened an issue to propose and discuss this change (#3426)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [X] Documentation

## Types of changes

- [X] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [X] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [X] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [X] I updated the documentation where necessary.

### CODE REVIEW

**Comment by lornajane** [comment]
Thanks!  These instructions enabled me to get much further with this setup.  It would be great if after the instructions to install the CLI, we also then gave instructions on how to run/check it?  The commands completed succesfully but it's not clear what I should do next (I will go and rummage around the docs, but we could make this easier I think)

File: ansible/README.md [hunk 1]
```diff
@@ -172,24 +172,42 @@ Use `ansible-playbook -i environments/<environment> openwhisk.yml` to avoid wipi
...
 
-All the binaries are generated and put under the folder of <openwhisk_cli_home>/bin. Then, use the following ansible command to configure the CLI installation mode:
+The binaries are generated and put into a tarball in the folder
+`../incubator-openwhisk-cli/release`.  Then, use the following ansible command
+to (re-)configure the CLI installation:
 
 ```
-ansible-playbook -i environments/<environment> openwhisk.yml -e cli_installation_mode=local -e openwhisk_cli_home=<openwhisk_cli_home>
+export OPENWHISK_ENVIRONMENT=local  # ... or whatever
+ansible-playbook -i environments/$OPENWHSIK_ENVIRONMENT edge.yml -e mode=clean
```
Comment (mdeuser) [inline ansible/README.md]: typo `OPENWHSIK` -> `OPENWHISK`

**Comment by jonpspri** [comment]
@lornajane A fair point.  I think that’s covered elsewhere in the ‘docs’ folder, but if you find the right place to go next, feel free to add a pointer to this file.

File: ansible/README.md [hunk 2]
```diff
@@ -172,24 +172,44 @@ Use `ansible-playbook -i environments/<environment> openwhisk.yml` to avoid wipi
...
-the installation mode to remote and downloads the binaries from the CLI [release page](https://github.com/apache/incubator-openwhisk-cli/releases), where OpenWhisk publishes the official `wsk` binaries.
+The mode "remote" means to download the `wsk` binaries from available web links.
+By default, OpenWhisk sets the installation mode to remote and downloads the
+binaries from the CLI
+[release page](https://github.com/apache/incubator-openwhisk-cli/releases),
+where OpenWhisk publishes the official `wsk` binaries.
 
-The mode "local" means to build and install the `wsk` binaries from local CLI project. You can download the source code
-of OpenWhisk CLI via [this link](https://github.com/apache/incubator-openwhisk-cli). Let's assume your OpenWhisk CLI home directory is <openwhisk_cli_home>. After you download the source code, use the gradle command to build the binaries:
+The mode "local" means to build and install the `wsk` binaries from local CLI
```
Comment (rabbah) [inline ansible/README.md]: from 'a' local CLI project.

Comment (jonpspri) [reply ansible/README.md]: Jeez.  It wasn’t even my typo — I just added a line feed.  Will attempt to correct, though.

Comment (rabbah) [reply ansible/README.md]: hehe I approved nonetheless and ready to merge.

**Pull request merged**